src/executables/elm_ats_api/elm_ats_api.cc

```diff
@@ -0,0 +1,55 @@
+
+#include "elm_ats_driver.hh"
+#include "elm_ats_api.h"
+
+#ifdef __cplusplus
+extern "C"{
+#endif
+
+// allocate, call constructor and cast ptr to opaque ELM_ATS_DRIVER
+ELM_ATS_DRIVER ats_create() {
```

Comment:
> is this void*?
Kind of. I used an opaque pointer method that's pretty common when creating portable C++ libraries. It's defined in the accompanying header file:
https://github.com/amanzi/ats/blob/a68f7b7bb32f4c991741f95df124bc187a77ba27/src/executables/elm_ats_api/elm_ats_api.h#L5-L16
To the C compiler and whatever linker the external program invokes, this is a typedef to a pointer to a forward declared dummy struct. It's like a void* with a bit of type safety. To the C++ compiler/linker, it's resolved as a ELM_ATSDriver*. 

I made the decision to hide the pointer behind the typedef because calling code should never dereference the ATS object. I think some people hate this, but it seems like the right call here.